src/openms/source/CHEMISTRY/ResidueDB.cpp

```diff
@@ -452,29 +481,36 @@ namespace OpenMS
 
   const Residue* ResidueDB::getModifiedResidue(const String& modification)
   {
+    OPENMS_NONUNIQUELOCK(mutex_rdb, lock)
+
     // terminal mods. don't apply to residue (side chain), so don't consider them:
     const ResidueModification& mod = ModificationsDB::getInstance()->getModification(modification, "", ResidueModification::ANYWHERE);
     return getModifiedResidue(getResidue(mod.getOrigin()), mod.getFullId());
   }
 
   const Residue* ResidueDB::getModifiedResidue(const Residue* residue, const String& modification)
```

Comment:
basically if this function would not modify ResidueDB and / or ModifcationDB then all access would be read-only and we would not need locks in the first place.